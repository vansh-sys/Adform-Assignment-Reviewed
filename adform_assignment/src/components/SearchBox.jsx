import React from 'react';
import '../index.css';
import Translations from '../translations/constants';

const SearchBox = (props) => (
  <input
    type="text"
    onChange={props.onSearch}
    name={Translations.lowerName}
    placeholder={Translations.searchPlaceholder}
    className="search-box"
  />
);

export default SearchBox;
