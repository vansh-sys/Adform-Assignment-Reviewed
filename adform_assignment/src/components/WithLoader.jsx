import React from 'react';
import Loader from 'react-loader-spinner';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import '../index.css';

const ComponentWithLoading = (Component) => ({ isLoading, ...props }) => {
  if (!isLoading) {
    return (
      <Component {...props} />
    );
  }
  return (
    <Loader
      type="ThreeDots"
      color="#00407f"
      height={100}
      width={100}
      className="loader"
    />
  );
};
export default ComponentWithLoading;
