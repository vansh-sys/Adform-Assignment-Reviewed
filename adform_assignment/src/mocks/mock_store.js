export const mockStore={
    userList : [],
    isLoading : true,
    campaignList:[
        {
           "id":1,
           "name":"Divavu",
           "startDate":"9/19/2019",
           "endDate":"3/9/2020",
           "Budget":88377,
           "userId":10
        },
        {
           "id":2,
           "name":"Jaxspan",
           "startDate":"11/21/2019",
           "endDate":"2/21/2020",
           "Budget":608715,
           "userId":6
        },
        {
           "id":3,
           "name":"Miboo",
           "startDate":"11/1/2019",
           "endDate":"6/20/2020",
           "Budget":239507,
           "userId":7
        },
        {
           "id":4,
           "name":"Trilith",
           "startDate":"8/25/2017",
           "endDate":"11/30/2019",
           "Budget":179838,
           "userId":1
        },
        {
           "id":5,
           "name":"Layo",
           "startDate":"11/28/2019",
           "endDate":"3/10/2020",
           "Budget":837850,
           "userId":9
        },
        {
           "id":6,
           "name":"Photojam",
           "startDate":"7/25/2017",
           "endDate":"6/23/2019",
           "Budget":858131,
           "userId":3
        },
        {
           "id":7,
           "name":"Blogtag",
           "startDate":"6/27/2018",
           "endDate":"1/15/2020",
           "Budget":109078,
           "userId":2
        },
        {
           "id":8,
           "name":"Rhyzio",
           "startDate":"10/13/2019",
           "endDate":"1/25/2020",
           "Budget":272552,
           "userId":4
        },
        {
           "id":9,
           "name":"Zoomcast",
           "startDate":"9/6/2019",
           "endDate":"11/10/2020",
           "Budget":301919,
           "userId":8
        },
        {
           "id":10,
           "name":"Realbridge",
           "startDate":"3/5/2018",
           "endDate":"10/2/2020 ",
           "Budget":505602,
           "userId":5
        }
     ],
     preparedList:[]
}