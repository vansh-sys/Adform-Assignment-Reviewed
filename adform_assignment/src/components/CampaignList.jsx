/* eslint-disable max-len */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { connect } from 'react-redux';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
import { AgGridReact } from 'ag-grid-react';
import DatePicker from 'react-date-picker';
import {
  fetchCampaigns, searchByName, prepareRowData, filterByDate, addNewCampaigns,
} from '../actions/campaignActions';
import StatusRederer from '../Renderer/StatusRenderer';
import '../index.css';
import addFormLogo from '../logos/addform.png';
import SearchBox from './SearchBox';
import Translations from '../translations/constants';
import selectCampaignUsers from '../Selectors/selectors';

export class CampaignList extends React.Component {
  constructor() {
    super();
    this.state = {
      isSearch: false,
      startDate: null,
      endDate: null,
      dateFilter: false,
    };
  }

  componentDidMount() {
    window.map = this;
    this.props.fetchCampaigns();
    // console.log("response",fetchResponse)
  }


  onSearch(e) {
    this.setState({ isSearch: true });
    this.props.searchByName(e.target.value);
    if (e.target.value === Translations.emptyString && this.state.dateFilter === false) { this.setState({ isSearch: false }); } else if (e.target.value === Translations.emptyString && this.state.dateFilter === true) { this.props.filterByDate(this.state.startDate, this.state.endDate, this.props.preparedList); }
  }


  onStartDateChange(date) {
    this.setState({
      startDate: date,
      dateFilter: true,
    }, this.props.filterByDate(date, this.state.endDate, this.props.campaignList));

    if (date === null && this.state.endDate === null && this.state.isSearch === false) {
      this.setState({
        dateFilter: false,
      });
    }
  }

  onEndDateChange(date) {
    this.setState({
      endDate: date,
      dateFilter: true,
    }, this.props.filterByDate(this.state.startDate, date, this.props.preparedList));


    if (this.state.startDate === null && date === null && this.state.isSearch === false) {
      this.setState({
        dateFilter: false,
      });
    }
  }

  render() {
    let rowData = [];
    // console.log("props",this.props.userList)
    if (this.props.userList.length > 0) {
      if (!this.state.isSearch && !this.state.dateFilter) {
        this.props.prepareRowData(this.props.userList);
      }
      rowData = this.props.campaignList;
    }
    const columnDefs = [
      { headerName: Translations.upperName, field: Translations.lowerName, filter: true },
      { headerName: Translations.upperUserName, field: Translations.lowerUserName },
      { headerName: Translations.upperStartDate, field: Translations.lowerStartDate },
      { headerName: Translations.upperEndDate, field: Translations.lowerEndDate },
      { headerName: Translations.upperStatus, field: Translations.lowerStatus, cellRenderer: 'statusRederer' },
      { headerName: Translations.upperBudget, field: Translations.lowerBudget },
    ];
    const frameworkComponents = {
      statusRederer: StatusRederer,
    };
    // this.gridApi.sizeColumnsToFit();
    return (
      <div>
        <header className="header">
          <img src={addFormLogo} alt="test logo" className="addFormLogo" />
        </header>

        <SearchBox onSearch={(e) => this.onSearch(e)} />

        <span className="dateFilter">Filter (by date) :</span>

        <span className="start-date">
          <DatePicker
            onChange={(e) => this.onStartDateChange(e)}
            name={Translations.startDate}
            dayPlaceholder={Translations.dateFormat}
            monthPlaceholder={Translations.monthFormat}
            yearPlaceholder={Translations.yearFormat}
            value={this.state.startDate}
            className="datePicker"
          />
        </span>

        <span className="end-date">
          <DatePicker
            onChange={(e) => this.onEndDateChange(e)}
            name={Translations.endDate}
            value={this.state.endDate}
            dayPlaceholder={Translations.dateFormat}
            monthPlaceholder={Translations.monthFormat}
            yearPlaceholder={Translations.yearFormat}
            className="datePicker"
          />
        </span>

        <br />
        <br />
        <div className="ag-theme-balham ag-grid-custom">
          <AgGridReact
            columnDefs={columnDefs}
            rowData={rowData}
            frameworkComponents={frameworkComponents}
            suppressHorizontalScroll
            false
            colResizeDefault
          />
        </div>

      </div>

    );
  }
}

const mapStateToprops = (state) => {
  const data = selectCampaignUsers(state);
  return data;
};

export default connect(
  mapStateToprops, {
    fetchCampaigns, searchByName, prepareRowData, filterByDate, addNewCampaigns,
  },
)(CampaignList);
