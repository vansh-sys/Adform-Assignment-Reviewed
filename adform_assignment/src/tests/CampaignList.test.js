import React from 'react';
import { shallow, configure } from 'enzyme';
import {CampaignList} from '../components/CampaignList';
import Adapter from 'enzyme-adapter-react-16';
import{mockStore} from '../mocks/mock_store';
configure({adapter: new Adapter()});

describe("CampaignList", () => {
  it("should render self & sub components", () => {
    const props = {
        userList : mockStore.userList,
        isLoading : mockStore.isLoading,
        campaignList : mockStore.campaignList,
        preparedList : mockStore.preparedList,
        fetchCampaigns : jest.fn()
    }  
    const wrapper = shallow(<CampaignList {...props}/>);
    //expect(wrapper.find('header').hasClass('header')).toBe(true)
    //expect(wrapper.find('img').hasClass('addFormLogo')).toBe(true)
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});