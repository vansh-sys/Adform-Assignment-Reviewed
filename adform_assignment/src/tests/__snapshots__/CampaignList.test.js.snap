// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignList should render self & sub components 1`] = `
Array [
  <div>
    <header
      className="header"
    >
      <img
        alt="test logo"
        className="addFormLogo"
        src="addform.png"
      />
    </header>
    <SearchBox
      onSearch={[Function]}
    />
    <span
      className="dateFilter"
    >
      Filter (by date) :
    </span>
    <span
      className="start-date"
    >
      <DatePicker
        calendarIcon={
          <svg
            className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        }
        className="datePicker"
        clearIcon={
          <svg
            className="react-date-picker__clear-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        }
        dayPlaceholder="dd"
        isOpen={null}
        monthPlaceholder="mm"
        name="startDate"
        onChange={[Function]}
        returnValue="start"
        value={null}
        yearPlaceholder="yyyy"
      />
    </span>
    <span
      className="end-date"
    >
      <DatePicker
        calendarIcon={
          <svg
            className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        }
        className="datePicker"
        clearIcon={
          <svg
            className="react-date-picker__clear-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        }
        dayPlaceholder="dd"
        isOpen={null}
        monthPlaceholder="mm"
        name="endDate"
        onChange={[Function]}
        returnValue="start"
        value={null}
        yearPlaceholder="yyyy"
      />
    </span>
    <br />
    <br />
    <div
      className="ag-theme-balham ag-grid-custom"
    >
      <AgGridReact
        columnDefs={
          Array [
            Object {
              "field": "name",
              "filter": true,
              "headerName": "Name",
            },
            Object {
              "field": "userName",
              "headerName": "User Name",
            },
            Object {
              "field": "startDate",
              "headerName": "Start Date",
            },
            Object {
              "field": "endDate",
              "headerName": "End Date",
            },
            Object {
              "cellRenderer": "statusRederer",
              "field": "status",
              "headerName": "Status",
            },
            Object {
              "field": "budget",
              "headerName": "Budget",
            },
          ]
        }
        frameworkComponents={
          Object {
            "statusRederer": [Function],
          }
        }
        rowData={Array []}
      />
    </div>
  </div>,
]
`;

exports[`SearchBox should render search component 1`] = `
Array [
  <input
    className="search-box"
    name="name"
    onChange={
      Object {
        "onSearch": [MockFunction],
      }
    }
    placeholder="Search By Name"
    type="text"
  />,
]
`;
